stan/math/fwd/functor/apply_scalar_unary.hpp

```diff
@@ -18,20 +18,23 @@ namespace math {
  * autodiff variable.
  */
 template <typename F, typename T>
-struct apply_scalar_unary<F, fvar<T> > {
+struct apply_scalar_unary<F, T, require_fvar_t<T>> {
   /**
    * Function return type, which is same as the argument type for
    * the function, <code>fvar&lt;T&gt;</code>.
    */
-  using return_t = fvar<T>;
+  using return_t = std::decay_t<T>;
 
   /**
    * Apply the function specified by F to the specified argument.
    *
    * @param x Argument variable.
    * @return Function applied to the variable.
    */
-  static inline return_t apply(const fvar<T>& x) { return F::fun(x); }
+  template <typename T2>
+  static inline auto apply(const T2& x) {
+    return F::fun(x);
+  }
```

Comment:
```suggestion
  static inline auto apply(T2&& x) {
    return F::fun(std::forward<T2>(x));
  }
```

Should this also be forwarding? As the downstream calls are forwarding their arguments to `apply_scalar_unary<*>::apply()`